pyparsing/helpers.py

```diff
@@ -46,7 +48,12 @@ def delimited_list(
     if not combine:
         delim = Suppress(delim)
 
-    delimited_list_expr = expr + ZeroOrMore(delim + expr)
+    if _min is None and _max is None:
+        delimited_list_expr = expr + ZeroOrMore(delim + expr)
+    else:
```

Comment:
Instead of this code, let's just value check the values, something like this:

    if None not in (min, max) and not (0 < min <= max):
        raise ValueError("min and max values must  be greater than 0, and min <= max")
    else:
        # one or both values are None, so no need to compare relative to each other
        if min is not None and min < 1:
            raise ValueError("min must be greater than 0")
        if max is not None and max < 1:
            raise ValueError("max must be greater than 0") 

Will probably want to add tests in unit_tests.py to make sure we get expected exceptions.

The value check now guards against getting min or max equal to 0, so can probably simplify to:

    _min = ... if min is None else min-1
    _max = ... if max is None else max-1